test/OrchardCore.Tests/Modules/OrchardCore.Tenants/ApiControllerTests.cs

```diff
@@ -0,0 +1,214 @@
+using OrchardCore.Data;
+using OrchardCore.Email;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Models;
+using OrchardCore.Environment.Shell.Removing;
+using OrchardCore.Setup.Services;
+using OrchardCore.Tenants;
+using OrchardCore.Tenants.Controllers;
+using OrchardCore.Tenants.Services;
+using OrchardCore.Tenants.ViewModels;
+
+namespace OrchardCore.Modules.OrchardCore.Tenants.Tests;
+
+public class ApiControllerTests
+{
+    private static readonly Dictionary<string, ShellSettings> _shellSettings = new();
+    private static readonly Mock<IClock> _clockMock = new();
+    private static readonly Dictionary<string, FeatureProfile> _featureProfiles = new()
+    {
+        { "Feature Profile", new FeatureProfile() }
+    };
```

Comment:
Why do these need to be static?

Reply:
Because they are used from within private static methods

Reply:
Those don't need to be static either. We don't need to keep state between the two tests (rather, that's usually a bad idea).

Reply:
I understand what you mean, that's why I clear the settings in the constructor to avoid unexpected behaviour

Reply:
Yes, but that shouldn't be necessary. It's a fix for an approach that's not needed.